docs/snippets/numpydocExample.py

```diff
@@ -0,0 +1,323 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""Example Python module with Numpydoc-formatted docstrings.
+
+This module demonstrates documentation written according to LSST DM's
+guidelines for `Documenting Python APIs with Docstrings`_. Docstrings have
+well-specified sections. This paragraph is considered an `Extended
+Summary`_. Permitted sections are listed in `Numpydoc Sections in Docstrings`_.
+You can't add arbitrary sections since they won't be parsed.
+
+Notes
+-----
+Usually we don't write extensive module docstrings. Focus the module docstring
+on information that a Stack developer needs to know when working inside that
+module. Module *users* typically won't see module docstrings (instead they will
+read module documentation topics written in the package's ``doc/`` directory).
+
+.. _`Documenting Python APIs with Docstrings`:
+   https://developer.lsst.io/docs/py_docs.html
+.. _`Extended Summary`:
+   https://developer.lsst.io/docs/py_docs.html#py-docstring-extended-summary
+.. _`Numpydoc Sections in Docstrings`:
+   https://developer.lsst.io/docs/py_docs.html#py-docstring-sections
+"""
+
+__all__ = ('MODULE_LEVEL_VARIABLE', 'moduleLevelFunction', 'exampleGenerator',
+           'ExampleClass', 'ExampleError')
+
+MODULE_LEVEL_VARIABLE = 12345
+"""Module level variable documented inline (`int`).
+
+The module variable's type is specified in the short summary, as shown above.
+Module variables (constants) can have extended descriptions, like this
+paragraph. For a complete list of sections permitted in constant docstrings see
+`Documenting Constants and Class Attributes`_.
+
+.. _`Documenting Constants and Class Attributes`:
+   https://developer.lsst.io/docs/py_docs.html#py-docstring-attribute-constants-structure
+"""
+
+
+def moduleLevelFunction(param1, param2=None, *args, **kwargs):
+    """Test that two parameters are not equal.
+
+    This is an example of a function docstring. Function parameters are
+    documented in the ``Parameters`` section. See *Notes* for the format
+    specification. See `Documenting Methods and Functions`_ for more
+    information about function docstrings.
+
+    Parameters
+    ----------
+    param1 : `int`
+        The first parameter. Note how the type is marked up with backticks.
+        This marks ``int`` as an API object so that Sphinx will attempt to
+        link to its reference documentation. You can do this for custom types
+        as well. You'll see an example in the *Returns* section.
+    param2 : `str`, optional
+        Optional arguments (those with defaults) always include the word
+        ``optional`` after the type info. See the `Parameters`_ section
+        documentation for details.
+    *args
+        Variable length argument list.
+    **kwargs
```

Comment:
If we're trying to give a complete example, shouldn't there be documentation on a keyword argument here?